### PR TITLE
supercronic: init at 0.2.27

### PR DESCRIPTION
**Pull request by NasaGeek**
###### Description of changes

Supercronic is a handy cron-alike that is well-suited for running in a container for the following reasons:
* Does not daemonize by default
* Preserves container environment for crontab
* Runs as non-root
* Logs crontab output to stdout

https://github.com/aptible/supercronic

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -10310,6 +10310,11 @@
     githubId = 6709831;
     name = "Jake Hill";
   };
+  nasageek = {
```
Comment (doronbehar) [inline maintainers/maintainer-list.nix:10313]: Should be a separate commit.

File: pkgs/tools/system/supercronic/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{ lib, buildGoModule, fetchFromGitHub, python3, bash, coreutils }:
```
Comment (doronbehar) [inline pkgs/tools/system/supercronic/default.nix:1]:
```suggestion
{ lib
, buildGoModule
, fetchFromGitHub
, python3
, bash
, coreutils
}:
```

File: pkgs/tools/system/supercronic/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+buildGoModule rec {
+  pname = "supercronic";
+  version = "0.2.1";
+
+  excludedPackages = [ "cronexpr/cronexpr" ];
+  checkInputs = [ python3 bash coreutils ];
+  vendorHash = "sha256-oSwow6Ajh4HWrr+hfjlJIp1DOBCvN1Khm4tseWokxoc=";
```
Comment (doronbehar) [inline pkgs/tools/system/supercronic/default.nix:9]: Could you put these attributes after `src`? It's common to have the `vendorHash` near the `src`'s hash - as they update together usually.

File: pkgs/tools/system/supercronic/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  src = fetchFromGitHub {
+    owner = "aptible";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-vaRoWXrCi7dLofluldiCT1yH0yCa4j1g+llv1MZDhOY=";
```
Comment (doronbehar) [inline pkgs/tools/system/supercronic/default.nix:15]: Nit:

```suggestion
    hash = "sha256-vaRoWXrCi7dLofluldiCT1yH0yCa4j1g+llv1MZDhOY=";
```

File: pkgs/tools/system/supercronic/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  postConfigure = ''
+    # There are tests that set the shell to various paths
+    substituteInPlace cron/cron_test.go --replace /bin/sh ${bash}/bin/sh
+    substituteInPlace cron/cron_test.go --replace /bin/false ${coreutils}/bin/false
+  '';
+
+  preCheck = ''
+    # There's a test that tries to run python
+    export PATH=${python3}/bin:$PATH
```
Comment (doronbehar) [inline pkgs/tools/system/supercronic/default.nix:26]: Why not add add `python3` to `checkInputs`?

**Review by doronbehar — CHANGES_REQUESTED** [review]

**Comment by NasaGeek** [comment]
Thanks very much for the feedback @doronbehar , I believe I've addressed it all.

File: pkgs/tools/system/supercronic/default.nix (hunk 5 above)
Comment (doronbehar) [reply pkgs/tools/system/supercronic/default.nix:34]: > It's already present in `checkInputs`, but I still had to explicitly add it to the `PATH` to get the test to pass. Let me know if there's something I'm missing here.

I think this will work if you'd use `nativeCheckInputs` instead. A while ago, `mkDerivation` and other builders started supporting only `nativeCheckInputs` instead of `checkInputs`.

**Review by doronbehar — CHANGES_REQUESTED** [review]
Sorry for the late reply to your PR update, I'm a bit busy nowadays...

The 2nd commit, should be:

```
supercronic: init at 0.2.27
```

Without the `v` prefix for the version.

**Review by doronbehar — APPROVED** [review]

**Comment by doronbehar** [comment]
Result of `nixpkgs-review pr 216746` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>supercronic</li>
  </ul>
</details>
